consul/state/catalog_test.go

```diff
@@ -27,42 +28,43 @@ func TestStateStore_EnsureRegistration(t *testing.T) {
 	s := testStateStore(t)
 
 	// Start with just a node.
+	nodeID := makeRandomNodeID(t)
 	req := &structs.RegisterRequest{
-		ID:      makeRandomNodeID(t),
-		Node:    "node1",
-		Address: "1.2.3.4",
-		TaggedAddresses: map[string]string{
-			"hello": "world",
-		},
-		NodeMeta: map[string]string{
-			"somekey": "somevalue",
-		},
+		ID:              nodeID,
+		Node:            "node1",
+		Address:         "1.2.3.4",
+		TaggedAddresses: map[string]string{"hello": "world"},
+		NodeMeta:        map[string]string{"somekey": "somevalue"},
 	}
-	nodeID := req.ID
 	if err := s.EnsureRegistration(1, req); err != nil {
 		t.Fatalf("err: %s", err)
 	}
 
 	// Retrieve the node and verify its contents.
 	verifyNode := func() {
+		node := &structs.Node{
+			ID:              nodeID,
+			Node:            "node1",
+			Address:         "1.2.3.4",
+			TaggedAddresses: map[string]string{"hello": "world"},
+			Meta:            map[string]string{"somekey": "somevalue"},
+			RaftIndex:       structs.RaftIndex{CreateIndex: 1, ModifyIndex: 1},
+		}
+
 		_, out, err := s.GetNode("node1")
 		if err != nil {
-			t.Fatalf("err: %s", err)
+			t.Fatalf("got err %s want nil", err)
 		}
-		if out.ID != nodeID ||
-			out.Node != "node1" || out.Address != "1.2.3.4" ||
-			len(out.TaggedAddresses) != 1 ||
-			out.TaggedAddresses["hello"] != "world" ||
-			out.Meta["somekey"] != "somevalue" ||
-			out.CreateIndex != 1 || out.ModifyIndex != 1 {
-			t.Fatalf("bad node returned: %#v", out)
+		if got, want := out, node; !verify.Values(t, "GetNode", got, want) {
```

Comment:
It's kind of a thing to remember to set every time. I was thinking that we could call `... !verify.Values(t, got, want)` and it would make a name based on where it was called from. I suppose we have the fail right below all these, though so we will get the info right there - making up a nice name for the structure will make the output more readable for a failure...you can probably disregard me here.